app/controllers/users_controller.rb

```diff
@@ -8,6 +8,22 @@ def show
       @location = Location.find(@user.location_id)
     end
     @favorite_discussions = @user.favorite_discussions
+    if @user.is_admin
+      @pending_map_users = User.where(map_approval_state: 0, on_map: true)
+    else
+      @pending_map_users = []
+    end
+  end
+
+  def approval_update
+    # batch update user's map_approval_state based on admin decision
```

Comment:
It might be a little picky, but move this comment before function declaration. This comment largely describes what function does and so I think it should go above function name.